src/index.ts

```diff
@@ -21,15 +21,20 @@ COPY ${source} ./`;
     dockerfile += `\nRUN ${buildCommand}`;
   }
 
-  fs.writeFile('Dockerfile', dockerfile, async (err: any) => {
+  fs.writeFile('Dockerfile.reproducible', dockerfile, async (err: any) => {
     if (err) throw err;
   });
 };
 
-export const createBuild = async (imageTag: string, dockerfile?: string) => new Promise((resolve, reject) => {
-  let dockerFilePath = '';
-  if (dockerfile) dockerFilePath = ` -f ${dockerfile}`;
-  const build = spawn(`docker build -t ${imageTag}${dockerFilePath} . --platform linux/amd64`, { shell: true });
+export const createBuild = async (
+  imageTag: string,
+  dockerfile: string = 'Dockerfile.reproducible',
+) => new Promise((resolve, reject) => {
+  // add ignore files to dockerignore
+  spawn(`cat .*ignore > ${dockerfile}.dockerignore`, { shell: true });
```

Comment:
Should probably use `spawnSync` to ensure no race condition.

Reply:
great catch! updated :)